src/render/index.js

```diff
@@ -1,25 +1,44 @@
 import { createElement } from '../utils/createElement';
 import DesktopRenderer from '../reconciler/';
 
+let connectToDevTools = () => {};
+
+// from jaredhanson/node-parent-require
+const prequire = function(id) {
+  for (let parent = module.parent; parent; parent = parent.parent) {
+    try {
+      return parent.require(id);
+    } catch (e) {}
+  }
+  throw new Error("Cannot find module '" + id + "' from parent");
+};
+
+try {
+  global.window = global;
+  global.window.WebSocket = prequire('ws');
+  connectToDevTools = prequire('react-devtools-core').connectToDevTools;
+} catch (e) {}
+
 export let ROOT_NODE = {};
 
 // Renders the input component
 function render(element) {
-  ROOT_NODE = createElement('ROOT');
-  const container = ROOT_NODE;
+  connectToDevTools();
+  DesktopRenderer.injectIntoDevTools({
```

Comment:
I need to prevent `injectIntoDevTools` from running in production

Reply:
It would probably be best to have it as an off-by-default option that the user can enable. I'm not sure what the best method for this would be.

Reply:
It will only try to connect to devtools if `react-devtools-core` is installed (so not in production). If react devtools aren't running it silently fails.
Is that acceptable?